lib/gui/etcher.js

```diff
@@ -40,14 +49,19 @@ const configUrl = settings.get('configUrl') || 'https://balena.io/etcher/static/
 const checkForUpdates = async (interval) => {
   // We use a while loop instead of a setInterval to preserve
   // async execution time between each function call
-  while (true) {
-    try {
-      const release = await autoUpdater.checkForUpdates()
-      if (release.updateInfo.stagingPercentage) {
-        await autoUpdater.downloadUpdate()
+  while (!packageUpdated) {
+    if (settings.get('updatesEnabled')) {
+      try {
+        const release = await autoUpdater.checkForUpdates()
+        const isOutdated = semver.compare(release.updateInfo.version, version) > 0
+        const shouldUpdate = parseInt(release.updateInfo.stagingPercentage, 10) > 0
```

Comment:
Are you sure this is necessary? Nothing I can find at https://www.electron.build/auto-update#staged-rollouts suggests you manually need to check the stagingPercentage value?

Reply:
@lurch It's because we manually check and then download, so it's an extra safety step before triggering the download for those connected with a data plan. One thing I didn't check is if `release.updateInfo.version` is unchanged in case no release is available (i.e. a newer release's `stagingPercentage` is 0), but I will